packages/@aws-cdk/aws-kinesisanalyticsv2/lib/flink-application.ts

```diff
@@ -0,0 +1,328 @@
+
+  /**
+   * The initial parallelism for the Flink application. Kinesis Data Analytics
+   * can stop the app, increase the parallelism, and start the app again if
+   * autoScalingEnabled is true (the default value).
+   *
+   * @default 1
+   */
+  readonly parallelism?: number;
+
+  /**
+   * The Flink parallelism allowed per Kinesis Processing Unit (KPU).
+   *
+   * @default 1
+   */
+  readonly parallelismPerKpu?: number
+
+  /**
+   * Determines if Flink snapshots are enabled.
+   *
+   * @default true
+   */
+  readonly snapshotsEnabled?: boolean;
+
+  /**
+   * Configuration PropertyGroups. You can use these property groups to pass
+   * arbitrary runtime configuration values to your Flink App.
+   */
+  readonly propertyGroups?: PropertyGroup[];
+
+  /**
+   * A role to use to grant permissions to your application. Prefer omitting
+   * this property and using the default role.
+   *
+   * @default - a new Role will be created
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Provide a RemovalPolicy to override the default.
+   *
+   * @default RemovalPolicy.DESTROY
+   */
+  readonly removalPolicy?: core.RemovalPolicy;
+}
+
+/**
+ * Attributes passed to the fromAttributes factory method.
+ */
+export interface Attributes {
+  readonly applicationName: string;
+  readonly applicationArn: string;
+}
+
+/**
+ * The L2 construct for Flink Kinesis Data Applications.
+ *
+ * @resource AWS::KinesisAnalyticsV2::Application
+ *
+ * @experimental
+ */
+export class FlinkApplication extends FlinkApplicationBase {
+  /**
+   * Import an existing Flink application, defined outside of CDK code.
+   */
+  public static fromAttributes(scope: Construct, id: string, attributes: Attributes): IFlinkApplication {
+    class Import extends FlinkApplicationBase {
+      // Imported flink applications have no associated role or grantPrincipal
+      public readonly role = undefined;
+      public readonly grantPrincipal = new iam.UnknownPrincipal({ resource: this });
+
+      public readonly applicationName = attributes.applicationName;
+      public readonly applicationArn = attributes.applicationArn;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly applicationArn: string;
+  public readonly applicationName: string;
+
+  // Role must be optional for JSII compatibility
+  public readonly role?: iam.IRole;
+
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  constructor(scope: Construct, id: string, props: FlinkApplicationProps) {
+    super(scope, id, { physicalName: props.applicationName });
+    validateFlinkApplicationProps(props);
+
+    this.role = props.role ?? new iam.Role(this, 'Role', {
+      assumedBy: new iam.ServicePrincipal('kinesisanalytics.amazonaws.com'),
+    });
+    this.grantPrincipal = this.role;
+
+    // Permit metric publishing to CloudWatch
+    this.role.addToPrincipalPolicy(new iam.PolicyStatement({
+      actions: ['cloudwatch:PutMetricData'],
+      resources: ['*'],
```

Comment:
To restrict this more I could add

```
      conditions: {
        StringEquals: {
          'cloudwatch:namespace': ['AWS/KinesisAnalytics', 'KinesisProducerLibrary'],
        },
      },
```

However I think Flink applications are more like Lambda functions than they are CloudWatch Canaries (the only thing in CDK I could find that restricts PutMetricData) because users may want to send arbitrary metrics to CloudWatch. I think it's more likely we'll frustrate users when they can't figure out why their custom metrics aren't showing up than delight them when we block some kind of unwanted metric creation.